### ISSUE TITLE
Incorrect variable value when using SNOPT with design variable 'scaler'

### ISSUE THREAD

**Issue by dsandler13** [OpenMDAO/OpenMDAO#2909]
### Description

Under very specific circumstances, the value of an input to an explicit component is being incorrectly set to its lower bound.
The following reproducing example is the paraboloid optimization problem from the documentation with modified initial values, bounds, and scalers.

Within the component, the input values should be unscaled. On the first evaluation, both x and y should therefore have their initial value of 1.0. This works as expected when pyOptSparse drives the optimization using SLSQP. However, when SNOPT is the optimizer, the value of x during the first evaluation is set to the variable's lower bound of 0.0 while the value of y is the correct value of 1.0.

I've tested a bunch of combinations of bounds, scalers, and initial values. It seems as if this bug(?) only occurs if the product of the initial value and the scaler is less than or equal to 1E-4, AND the lower bound is greater than or equal to zero. As far as I can tell, a similar issue is not present for the upper bound.

The combination of conditions required is baffling. I'm not sure if this is a problem with OpenMDAO, pyOptSparse, or SNOPT itself. Any assistance would be greatly appreciated!

### Example

```python
import openmdao.api as om

class Paraboloid(om.ExplicitComponent):
    """
    Evaluates the equation f(x,y) = (x-3)^2 + xy + (y+4)^2 - 3.
    """
    def setup(self):
        self.add_input('x')
        self.add_input('y')

        self.add_output('f_xy')

    def setup_partials(self):
        # Finite difference all partials.
        self.declare_partials('*', '*', method='fd')

    def compute(self, inputs, outputs):
        """
        f(x,y) = (x-3)^2 + xy + (y+4)^2 - 3

        Minimum at: x = 6.6667; y = -7.3333
        """
        x = inputs['x']
        y = inputs['y']
        print(f'x: {x}')
        print(f'y: {y}')
        outputs['f_xy'] = (x - 3.0)**2 + x * y + (y + 4.0)**2 - 3.0
        print(f'f_xy: {outputs["f_xy"]}')

# build the model
prob = om.Problem()
prob.model.add_subsystem('parab', Paraboloid(), promotes_inputs=['x', 'y'])

# Design variables 'x' and 'y' span components, so we need to provide a common initial
# value for them.
prob.model.set_input_defaults('x', 1)
prob.model.set_input_defaults('y', 1)

# setup the optimization
prob.driver = om.pyOptSparseDriver()
prob.driver.options['optimizer'] = 'SNOPT'
#prob.driver.options['optimizer'] = 'SLSQP'

prob.model.add_design_var('x', lower=0.0, upper=50000,scaler=1/10000)
prob.model.add_design_var('y', lower=0.0, upper=50000,scaler=1)
prob.model.add_objective('parab.f_xy')

prob.setup()
prob.run_driver();
```

### OpenMDAO Version

3.26.0

### Relevant environment information

I am using Python 3.9.2 along with the following packages and dependencies:

numpy==1.21.2
pyoptsparse==2.9.0
scipy==1.9.3

SNOPT Version 7.7.0

**Comment by robfalck** [comment]
We've spent some time tracking this one down, and we've confirmed that we're handing SNOPT the design variables and their bounds in a correctly scaled state.  Once SNOPT begins its execution, we don't have complete control over the values of the design variables that it chooses to analyze.  I agree that it's concerning that SNOPT does this but I'm not sure theres a simple way to avoid this.

The value of the scaling definitely has an impact, and setting the scaler to something smaller (1/1000.) will result in the first value of x being 1.0, as expected.
